src/System.Console/src/System/Console.cs

```diff
@@ -73,6 +90,41 @@ private static TextWriter CreateOutputWriter(Stream outputStream)
                     leaveOpen: true) { AutoFlush = true });
         }
 
+        private static bool _stdInRedirectQueried = false;
+        private static bool _stdOutRedirectQueried = false;
+        private static bool _stdErrRedirectQueried = false;
+
+        private static bool _isStdInRedirected;
+        private static bool _isStdOutRedirected;
+        private static bool _isStdErrRedirected;
+
+        public static bool IsInputRedirected
+        {
+            get
+            {
+                return Volatile.Read(ref _stdInRedirectQueried) ? _isStdInRedirected
+                    : EnsureInitialized(ref _isStdInRedirected, ref _stdInRedirectQueried, () => { return ConsolePal.IsInputRedirectedCore(); });
```

Comment:
I am not sure how StrongBox<T> is implemented but  won't we pay unboxing penalty with it every time?
Sorry, I am missing your concern here, after the first call to EnsureInitialized _stdXXXRedirectQueried should be set to true and so every subsequent call should return stdXXXRedirect directly w/o calling the initialization again, irrespective of it being true or false. Is not it?